src/openforms/contrib/kadaster/api/views.py

```diff
@@ -0,0 +1,111 @@
+
+from openforms.api.serializers import ExceptionSerializer, ValidationErrorSerializer
+from openforms.api.views.mixins import ListMixin
+from openforms.contrib.kadaster.models import KadasterApiConfig
+from openforms.submissions.api.permissions import AnyActiveSubmissionPermission
+
+from .serializers import MapSearchSerializer
+
+logger = logging.getLogger(__name__)
+
+
+@extend_schema(
+    summary=_("List BAG address suggestions."),
+    parameters=[
+        OpenApiParameter(
+            "q",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description=_("The search query we send to the pdok locatieserver api."),
+            required=True,
+        )
+    ],
+    responses={
+        200: MapSearchSerializer(many=True),
+        status.HTTP_400_BAD_REQUEST: ValidationErrorSerializer,
+        status.HTTP_403_FORBIDDEN: ExceptionSerializer,
+    },
+)
+class MapSearchView(ListMixin, APIView):
+    authentication_classes = ()
+    permission_classes = [AnyActiveSubmissionPermission]
+    serializer_class = MapSearchSerializer
+
+    def get_objects(self):
+        config = KadasterApiConfig.get_solo()
+        if not config.kadaster_service:
+            raise ValidationError(
+                {"invalid_params": _("No service defined for the Kadaster service.")}
+            )
+
+        query = self.request.GET.get("q")
+        if not query:
+            raise ValidationError({"invalid_params": _("Missing query parameter 'q'")})
+
+        kadaster_client = config.kadaster_service.build_client()
+        query_params = {"q": f"{query}"}
+
+        bag_response = dict()
+        try:
+            bag_response = kadaster_client.operation(
+                operation_id="free",
+                data=None,
+                url="v3_1/free",
+                method="GET",
+                request_kwargs=dict(
+                    params=query_params,
+                ),
+            )
+        except (ClientError, requests.RequestException):
+            logger.exception("couldn't retrieve pdok locatieserver data")
+
+        if not (
+            (response := bag_response.get("response"))
+            and (docs := response.get("docs"))
+        ):
+            # no response with docs: return empty
+            return []
+
+        def parse_doc(doc: dict) -> dict | None:
+            """Parse a doc from pdok response into location"""
+            try:
+                lng, lat = fromstr(doc.get("centroide_ll"))
+            except (ValueError, TypeError):
+                logger.info(
+                    f"Malformed centroide_ll in pdok locatieserver response: {doc}"
+                )
+                return
+
+            try:
+                x, y = fromstr(doc.get("centroide_rd"))
+            except (ValueError, TypeError):
+                logger.info(
+                    f"Malformed centroide_rd in pdok locatieserver response: {doc}"
+                )
+                return
```

Comment:
This can be cleaned up substantially:

1. There is no need for this function to be defined runtime - move it to the module level
2. The GEO data parsing is a repetition that can be its own function

```py
def parse_coordinates(doc: dict, key: str) -> tuple[float, float] | None:
    if not (wkt_value := doc.get(key)):
        return None
    try:
        coord0, coord1 = fromstr(wkt_value)  # we expect a two-tuple, and not any arbitrary shape
    except (ValueError, TypeError):
        logger.info("Malformed %s in location server response: %r", key, doc)
        return None
    return coord0, coord1


@dataclass
class RD:
    x: float
    y: float

    @classmethod
    def fromtuple(cls, value: tuple[float, float]):
        x, y = value
        return cls(x=x, y=y)


@dataclass
class LatLng:
    lat: float
    lng: float

    @classmethod
    def fromtuple(cls, value: tuple[float, float]):
        lat, lng = value
        return cls(lat=lat, lng=lng)


@dataclass
class Location:
    label: str
    lat_lng: LatLng
    rd: RD
    

def process_search_result(doc: dict) -> Location | None:
    label = doc.get("weergavenaam", "")
    lat_lng = parse_coordinates(doc, "centroide_ll")
    x_y = parse_coordinates(doc, "centroid_rd")
    if not lat_lng or not x_y:
        return None
    return Location(
        label=label, 
        lat_lng=LatLng.fromtuple(lat_lng),
        rd=RD.fromtuple(x_y),
    )
```

I have some some doubts about filtering out search results that are missing RD though :thinking: why would those not be displayed?